src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIStructure.java

```diff
@@ -901,43 +1006,87 @@ public void reduceNeededResources(final Block block)
         }
     }
 
-    private void spawnEntity(@Nullable final Entity entity)
+    /**
+     * Get the entity of an entityInfo object.
+     * @param entityInfo the input.
+     * @return the output object or null.
+     */
+    @Nullable
+    private Entity getEntityFromEntityInfoOrNull(Template.EntityInfo entityInfo)
     {
-        if (entity != null && job instanceof JobBuilder)
+        try
         {
-            final BlockPos pos = ((JobBuilder) job).getStructure().getOffsetPosition();
-
-            if (entity instanceof EntityHanging)
-            {
-                @NotNull final EntityHanging entityHanging = (EntityHanging) entity;
+            return EntityList.createEntityFromNBT(entityInfo.entityData, world);
+        }
+        catch (Exception e)
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Catch a list of specific exception subtypes instead. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS2221)